components/team-codes/Create.js

```diff
@@ -1,74 +1,99 @@
+  Text,
+  View,
+  StyleSheet,
+  TouchableOpacity,
+  TextInput
+} from "react-native";
+import KitButton from "../KitButton";
+import KitText from "../KitText";
+import Colors from "../../constants/Colors";
+import FontStyles from "../../constants/FontStyles";
+import { createTeam } from "../../utils/db/teams";
 
 export function Create(props) {
-  const placeholderImage = require('../../assets/images/40by40.png')
+  const [teamName, setTeamName] = useState("");
+  const [teamCode, setTeamCode] = useState("");
+
+  const placeholderImage = require("../../assets/images/40by40.png");
 
   return (
     <View style={styles.background}>
       <View style={styles.main}>
-        <KitButton 
-          style={{maxHeight: 40, marginLeft: 0, marginRight: 0, justifyContent: "flex-start"}} 
-          image={placeholderImage}>
-        </KitButton>
+        <KitButton
+          style={{
+            maxHeight: 40,
+            marginLeft: 0,
+            marginRight: 0,
+            justifyContent: "flex-start"
+          }}
+          image={placeholderImage}
+        ></KitButton>
 
-        <View style={{flex:2, justifyContent:"center"}}>
-          <KitText 
-            style={{}} 
-            size={18} 
-            color={Colors.KIT_DARKEST_BLACK} 
-            fontWeight={FontStyles.FONT_WEIGHT_LIGHT}>
-              Set Team Name
+        <View style={{ flex: 2, justifyContent: "center" }}>
+          <KitText
+            style={{}}
+            size={18}
+            color={Colors.KIT_DARKEST_BLACK}
+            fontWeight={FontStyles.FONT_WEIGHT_LIGHT}
+          >
+            Set Team Name
           </KitText>
 
           {/* Evan TODO: Make this input a separate compo */}
-          <TextInput style={styles.textInput} 
-            placeholder="Ex) Team One" 
-            onChangeText={props.onChangeText} 
-            value={props.teamNameText} />
+          <TextInput
+            style={styles.textInput}
+            placeholder="Ex) Team One"
+            onChangeText={setTeamName}
+            value={teamName}
+          />
         </View>
 
-        <View style={{flex:4,justifyContent:"flex-start"}}>
-          <KitText 
-            style={{}} 
-            fontWeight={FontStyles.FONT_WEIGHT_BOLD} 
-            color={Colors.KIT_DARKEST_BLACK} 
-            size={53}>
-              47VK0A
-          </KitText>
-
-          <KitButton 
-            style={{button: styles.button}} 
-            onPress={()=>alert("pressed!")} 
-            buttonBackgroundColor={Colors.KIT_GREEN} 
-            buttonTextColor={Colors.KIT_WHITE} 
-            buttonFontWeight={FontStyles.FONT_WEIGHT_MEDIUM} 
-            buttonFontSize={18} 
-            buttonTextStyle={{}}>
-              SHARE
+        <View style={{ flex: 4, justifyContent: "flex-start" }}>
+          <KitButton
+            style={{ button: styles.button }}
+            onPress={async () => {
+              const teamCode = await createTeam(teamName, "TODO: add userid");
```

Comment:
The TODO is still WIP right?

If we are using firebase authentication, the following might work:
`var user = firebase.auth().currentUser;`
https://firebase.google.com/docs/auth/web/manage-users

Reply:
thanks. i am thinking about just using that in teams.js instead of passing the user id into createTeam since you should always pass in yourself anyway.